### PR TITLE
examples for some blueprint uses in docs

### PR DESCRIPTION
**Pull request by berkio3x**
Adding examples for the usage of Blueprints as templates & subdomains. #3675
Link to any relevant issues or pull requests.

### CODE REVIEW

File: docs/blueprints.rst [hunk 1]
```diff
@@ -20,9 +20,10 @@ Why Blueprints?
 
 Blueprints in Flask are intended for these cases:
 
-* Factor an application into a set of blueprints.  This is ideal for
-  larger applications; a project could instantiate an application object,
-  initialize several extensions, and register a collection of blueprints.
+* Factor an application into a modular set of related logical components,
+  ideal for larger applications. A project could instantiate an application
+  object , register a collection of blueprints on the app. This makes
```
Comment (henrycjc) [inline docs/blueprints.rst:25]: `object ,` has a trailing space

File: docs/blueprints.rst [hunk 2]
```diff
@@ -121,6 +122,50 @@ On top of that you can register blueprints multiple times though not every
 blueprint might respond properly to that.  In fact it depends on how the
 blueprint is implemented if it can be mounted more than once.
 
+
+Using Blueprint to provide templates
+------------------------------------
+Register a blueprint multiple times on an application with different URL rules::
+
+  from flask import Flask , Blueprint, url_for , Request, jsonify
```
Comment (henrycjc) [inline docs/blueprints.rst:130]: Trailing spaces here too

File: docs/blueprints.rst (hunk 1 above)
Comment (henrycjc) [reply docs/blueprints.rst:25]: How about this:
`Architect a Flask app into a modular set of logically related components (ideal for larger apps). A project could instantiate a Flask app object, and then register a collection of blueprints on this instance. The blueprints can be reused, meaning your code can be made easier to maintain.`

File: docs/blueprints.rst [hunk 3]
```diff
@@ -121,6 +122,50 @@ On top of that you can register blueprints multiple times though not every
...
+  app.register_blueprint(library, url_prefix="/biology")
+  app.register_blueprint(library, url_prefix="/geography")
+
+
+Using Blueprint for subdomains
+---------------------------------
+Blueprints can be used to create subdomains::
+
+  from flask import Blueprint
+  from flask import Flask
```
Comment (henrycjc) [inline docs/blueprints.rst:156]: `from flask import Flask, Blueprint`

**Comment by davidism** [comment]
Thanks for working on this, but I don't think I'm going to merge these. The two examples are fairly large code blocks, but don't really show off the feature they're about and have no explanation. The first example isn't very useful either, you would use a variable part, not multiple hard coded prefixes.

**Pull request closed without merging**